paasta_tools/kubernetes/remote_run.py

```diff
@@ -141,53 +159,75 @@ def remote_run_start(
 
 
 def remote_run_ready(
-    service: str, instance: str, cluster: str, job_name: str
+    service: str,
+    instance: str,
+    cluster: str,
+    job_name: str,
+    user: str,
+    is_toolbox: bool,
 ) -> RemoteRunOutcome:
     """Check if remote-run pod is ready
 
     :param str service: service name
     :param str instance: service instance
     :param str cluster: paasta cluster
     :param str job_name: name of the remote-run job to check
+    :param bool is_toolbox: requested job is for a toolbox container
     :return: job status, with pod info
     """
     kube_client = KubeClient()
 
     # Load the service deployment settings
-    deployment_config = load_eks_service_config(service, instance, cluster)
+    deployment_config = (
+        generate_toolbox_deployment(service, cluster, user)
+        if is_toolbox
+        else load_eks_service_config(service, instance, cluster)
+    )
     namespace = deployment_config.get_namespace()
 
     pod = find_job_pod(kube_client, namespace, job_name)
     if not pod:
         return {"status": 404, "message": "No pod found"}
     if pod.status.phase == "Running":
-        return {
+        result: RemoteRunOutcome = {
             "status": 200,
             "message": "Pod ready",
             "pod_name": pod.metadata.name,
             "namespace": namespace,
         }
+        if is_toolbox:
```

Comment:
```suggestion
        # NOTE: we only do this for toolbox containers since we need to SSH in rather than exec in to maintain user identities
        # i.e., being `nobody` is fine in a normal remote-run, but in a toolkit, there's likely tooling that will require knowing the real user
        if is_toolbox:
```

Reply:
(iirc, this is why the toolboxes are SSH-able - but please correct me if i'm wrong)

Reply:
(mostly just wanna make sure no one tries to simplify things in the future down to always execing)